src/NLog.Web/Internal/HttpCookieCollectionValues.cs

```diff
@@ -85,5 +85,95 @@ internal static class HttpCookieCollectionValues
+        {
+            if (cookie.Values.Count > 1)
+            {
+                // Split multi-valued cookie, as allowed for in the HttpCookie API for backwards compatibility with classic ASP
+                var isFirst = true;
+                foreach (var multiValueKey in cookie.Values.AllKeys)
+                {
+                    var cookieKey = multiValueKey;
+                    if (isFirst)
+                    {
+                        cookieKey = cookie.Name;
+                        isFirst = false;
+                    }
+                    yield return new HttpCookie(cookieKey, cookie.Values[multiValueKey])
+                    {
+                        Domain = cookie.Domain,
+                        Path = cookie.Path,
+                        Expires = cookie.Expires,
+                        Secure = cookie.Secure,
+                        HttpOnly = cookie.HttpOnly
+                    };
+                }
+            }
+            else
+            {
+                yield return cookie;
+            }
+        }
+
+        internal static IEnumerable<HttpCookie> GetVerboseCookieValues(HttpCookieCollection cookies, List<string> cookieNames, HashSet<string> excludeNames, bool expandMultiValue)
+        {
+            if (cookieNames?.Count > 0)
+            {
+                return GetCookieVerboseNameValues(cookies, cookieNames, expandMultiValue);
+            }
+            else
+            {
+                return GetCookieVerboseAllValues(cookies, excludeNames, expandMultiValue);
+            }
+        }
+
+        private static IEnumerable<HttpCookie> GetCookieVerboseNameValues(HttpCookieCollection cookies, List<string> cookieNames, bool expandMultiValue)
+        {
+            foreach (var cookieName in cookieNames)
+            {
+                var httpCookie = cookies[cookieName];
+                if (httpCookie == null)
+                {
+                    continue;
+                }
+
+                if (expandMultiValue)
+                {
+                    foreach (var cookie in GetCookieMultiVerboseValues(httpCookie))
+                        yield return cookie;
+                }
+                else
+                {
+                    yield return httpCookie;
+                }
+            }
+        }
+
+        private static IEnumerable<HttpCookie> GetCookieVerboseAllValues(HttpCookieCollection cookies, HashSet<string> excludeNames, bool expandMultiValue)
+        {
+            bool checkForExclude = excludeNames?.Count > 0;
+            foreach (string cookieName in cookies.Keys)
+            {
+                if (checkForExclude && excludeNames.Contains(cookieName))
+                    continue;
+
+                var httpCookie = cookies[cookieName];
+                if (httpCookie == null)
+                {
+                    continue;
+                }
+
+                if(expandMultiValue)
```

Comment:
Could it become if (expandMultiValue && httpCookie.Values.Count > 1)

Reply:
Changed as requested